client.go

```diff
@@ -0,0 +1,221 @@
+package dalga
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/dalga/internal/jobmanager"
+	"github.com/cenkalti/dalga/internal/server"
+	"github.com/cenkalti/dalga/internal/table"
+)
+
+var (
+	ErrNotExist = table.ErrNotExist
+)
+
+type ClientOpt func(c *Client)
+
+func WithClient(clnt *http.Client) ClientOpt {
+	return func(c *Client) {
+		c.clnt = clnt
+	}
+}
+
+type Client struct {
+	clnt    *http.Client
+	BaseURL string
+}
+
+func NewClient(baseURL string, opts ...ClientOpt) *Client {
+	c := &Client{
+		BaseURL: strings.TrimSuffix(baseURL, "/"),
+		clnt:    http.DefaultClient,
+	}
+	for _, o := range opts {
+		o(c)
+	}
+	return c
+}
+
+func (clnt *Client) Get(ctx context.Context, path, body string) (*Job, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, clnt.jobURL(path, body), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clnt.clnt.Do(req)
+	if err != nil {
+		select {
+		case <-ctx.Done():
+			err = ctx.Err()
+		default:
+		}
```

Comment:
Is this `select` block really required? What happens if we omit it?

Reply:
This was actually inspired by https://github.com/golang/net/blob/master/context/ctxhttp/ctxhttp.go#L28
```
// If we got an error, and the context has been canceled,
// the context's error is probably more useful.
```
I don't think it's vital, but it seemed like a good point.